chalice/package.py

```diff
@@ -625,9 +625,15 @@ def _generate_s3bucketnotification(self, resource, template):
 
         # we use the bucket name here because we need to aggregate
         # all the notifications subscribers for a bucket.
+        # Due to cyclic references to buckets created in terraform
+        # we also try to detect and resolve.
+        if '{aws_s3_bucket.' in resource.bucket:
```

Comment:
I think that having this quick check here is fine for now. But I have a feeling we will have to come up with a more general way to handle terraform references for resource names. Fortunately, we can rely on the name of the function for a lot of these terraform resources instead of the AWS resource the function is hooked up to, but people in the future may want to override the name of the lambda function (i.e. with the `name` parameter) with a terraform reference.

Reply:
this is basically the only place we use a user supplied var name in the resource, as we're using the user input for the name instead of the chalice resource name for it (aka python function name) to enable bucket level grouping here for the multiple lambda functions on a bucket scenario you had posited previously, else we're using the chalice resource name for all other resources.

Reply:
wrt to name parameter, is there a scenario where that would include a terraform reference ?

Reply:
That is what I'm trying to think if it it would make sense ever to include a terraform reference in the name. So far I have not come up with any use case.